src/ui/Base/Select.tsx

```diff
@@ -0,0 +1,87 @@
+import {
+  ChangeEvent,
+  FocusEvent,
+  forwardRef,
+  Fragment,
+  PropsWithChildren,
+  ReactNode,
+  Ref,
+} from 'react';
+import { Listbox } from '@headlessui/react';
+import SelectIcon from '@public/icons/select-icon.svg';
+import { FieldError } from 'react-hook-form';
+
+interface SelectProps {
+  name?: string;
+  onChange?: (e: ChangeEvent<HTMLInputElement>) => void;
+  onBlur?: (e: FocusEvent<HTMLInputElement>) => void;
+  placeholder?: string;
+  error?: FieldError;
+}
+
+const Select = (
+  {
+    placeholder = '',
+    error,
+    children,
+    ...props
+  }: PropsWithChildren<SelectProps>,
+  ref: Ref<HTMLElement>
+) => {
+  const borderColor = error
+    ? 'border-warning-800'
+    : 'border-black-400 data-[headlessui-state=open]:border-main-900';
+
+  return (
+    <div className="flex flex-col gap-[0.5rem] text-sm">
+      <Listbox as="div" {...props} ref={ref} className="relative">
+        <Listbox.Button as={Fragment}>
+          {({ value }) => (
+            <button
+              className={`relative flex w-full items-center rounded-[0.5rem] border-[0.05rem] px-[1rem] py-[1.3rem] ${borderColor}`}
+            >
+              {value ? (
+                <div className="min-h-[2.1rem] w-full pr-[2.35rem] text-left">
+                  {value}
+                </div>
+              ) : (
+                <div className="min-h-[2.1rem] w-full pr-[2.35rem] text-left text-placeholder">
+                  {placeholder}
+                </div>
+              )}
+              <SelectIcon className="absolute right-[1.35rem]" />
+            </button>
+          )}
+        </Listbox.Button>
+        {children && (
+          <div className="absolute z-10 w-full">
+            <Listbox.Options className=" my-[0.5rem] rounded-[0.5rem] bg-white shadow-md">
+              {children}
+            </Listbox.Options>
+          </div>
+        )}
```

Comment:
`Listbox.Options` 컴포넌트에서 `Listbox`의 상태가 open이 아니면 내부적으로 ul 태그를 렌더링하지 않는 것 같아요. 그리고 `div` 태그에 적용된 스타일을 `Listbox.Options`에 줘도 의도한대로 동작할 것 같아요!

`Listbox.Options` 스타일에 `max-height` 속성을 줘서 overflow가 되는 경우 scroll이 되게 하는건 어떨까요? (docs 보고 작성한 코드라 참고만 해주세요!)

이렇게 작성하면 더 간단해질 것 같은데 어떠신가요 민종님?!

```suggestion
        <Listbox.Options className="absolute z-10 mt-1 max-h-60 w-full overflow-auto rounded-[0.5rem] bg-white py-1 shadow-md focus:outline-none sm:text-sm">
          {children}
        </Listbox.Options>
```

Reply:
아 그렇네요 children 으로 조건 렌더링을 하는게 의미가 없군요! 👍 
제안해주신 코드도 괜찮은 것 같아요. 그런데 기본 select 태그를 활용하는 쪽으로 바꿔야 할 것 같아요 ㅎㅎ..